app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/TransferSendManager.kt

```diff
@@ -78,72 +76,51 @@ class TransferSendManager @Inject constructor(
     private suspend fun sendTransfer(uploadSessionUuid: String) {
         _sendStatus.value = SendStatus.Pending
 
-        withIntegrityToken(
-            onSuccess = { attestationToken ->
-                runCatching {
-                    uploadManager.initUploadSession(
-                        attestationHeaderName = AppIntegrityManager.ATTESTATION_TOKEN_HEADER,
-                        attestationToken = attestationToken,
-                    )!! // TODO: Handle ContainerErrorsException here
-                    uploadWorkerScheduler.scheduleWork(uploadSessionUuid)
-                    _sendStatus.update {
-                        val totalSize = importationFilesManager.importedFiles.value.sumOf { it.fileSize }
-                        SendStatus.Success(totalSize)
-                    }
-                }.onFailure { exception ->
-                    SentryLog.e(TAG, "Failed to start the upload", exception)
-                    _sendStatus.update { SendStatus.Failure }
-                }
-            },
-            onRefused = { _sendStatus.update { SendStatus.Refused } },
-        )
+        runCatching {
+            val attestationToken = getAttestationToken()
+            uploadManager.initUploadSession(
+                attestationHeaderName = AppIntegrityManager.ATTESTATION_TOKEN_HEADER,
+                attestationToken = attestationToken,
+            )!! // TODO: Handle ContainerErrorsException here
+            uploadWorkerScheduler.scheduleWork(uploadSessionUuid)
+            _sendStatus.update {
+                val totalSize = importationFilesManager.importedFiles.value.sumOf { it.fileSize }
+                SendStatus.Success(totalSize)
+            }
+        }.onFailure { exception ->
+            if (exception !is NetworkException && exception !is KmpNetworkException) {
+                SentryLog.e(TAG, "Failed to start the upload", exception)
+            }
+            if (exception is IntegrityException) {
+                _sendStatus.update { SendStatus.Refused }
+            } else {
+                _sendStatus.update { SendStatus.Failure }
+            }
+        }
     }
 
     //region App Integrity
-    private suspend inline fun withIntegrityToken(onSuccess: (attestationToken: String) -> Unit, onRefused: () -> Unit = {}) {
-        var attestationToken: String? = null
-
-        coroutineScope {
-            appIntegrityManager.getChallenge(
-                onSuccess = { launch { attestationToken = requestAppIntegrityToken(appIntegrityManager) } },
-                onFailure = {},
-            )
-        }
-
-        attestationToken?.let(onSuccess) ?: onRefused()
+    /** Handle [IntegrityException] and all the different exceptions that can be thrown */
```

Comment:
This comment is outdated as the error handling is on the outside of this function